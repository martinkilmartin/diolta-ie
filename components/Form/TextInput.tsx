type Props = {
  title?: string
  placeholder: string
}

const TextInput = ({ title, placeholder }: Props): JSX.Element => (
  <div className="form-control">
    <label className="input-group">
      {title && <span>{title}</span>}
      <input
        type="text"
        placeholder={placeholder}
        className="input input-bordered input-primary text-base-content"
      />
    </label>
  </div>
)

export default TextInput
