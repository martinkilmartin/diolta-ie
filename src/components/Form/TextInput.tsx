type Props = {
  title?: string
  type?: string
  name?: string
  id?: string
  minLength?: number
  maxLength?: number
  placeholder: string
}

const TextInput = ({
  title,
  type = 'text',
  name,
  minLength = 1,
  maxLength = 1000,
  id,
  placeholder,
}: Props): JSX.Element => (
  <div className="form-control">
    {title && (
      <label className="label">
        <span>{title}</span>{' '}
      </label>
    )}
    <input
      type={type}
      placeholder={placeholder}
      className="input input-primary input-bordered"
      name={name}
      id={id}
      minLength={minLength}
      maxLength={maxLength}
      required
    />
  </div>
)

export default TextInput
