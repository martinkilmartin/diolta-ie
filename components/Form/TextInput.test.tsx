import renderer from 'react-test-renderer'
import TextInput from './TextInput'

test('TextInput displays label and text-input', () => {
  const title = 'Email'
  const placeholder = 'info@sionnach.solutions'
  const textInputRenderer = renderer.create(
    <TextInput title={title} placeholder={placeholder} />
  )
  const tree = textInputRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const textInputInstance = textInputRenderer.root
  expect(textInputInstance.findByType('span').children[0]).toEqual(title)
  expect(
    textInputInstance.findByProps({
      className: 'input input-bordered input-primary text-base-content',
    }).props.placeholder
  ).toEqual(placeholder)
})
