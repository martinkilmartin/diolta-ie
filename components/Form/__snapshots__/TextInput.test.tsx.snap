// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput displays label and text-input 1`] = `
<div
  className="form-control"
>
  <label
    className="input-group"
  >
    <span>
      Email
    </span>
    <input
      className="input input-bordered input-primary text-base-content"
      placeholder="info@sionnach.solutions"
      type="text"
    />
  </label>
</div>
`;
